### ISSUE TITLE
Out of Range error - TypeFilter = 2

### ISSUE THREAD

**Issue by Miiraak** [Miiraak/ProxySharp#1]
When `typeFilter` is set to 2, the used var `Indexes` is used to store the proxes indexes matched by filters values.
The variable is passed through two methodes. One for `value1` and another one for `value2` (country and port in our case).

When the value is passed to `CountryFilter`, all working great, right index are added to `indexes`. 
The problem come when indexes will be added with the second methode.

If the value in `.item3` equal `value2` it will be added to the `Indexes` even if the `.item2` not match the filter.
We need to do the verification on already gathered indexes proxies.

Can't change the methode for port cause is used in another methode or functions.

**Comment by Miiraak** [comment]
```csharp
foreach (index in Indexes)
{
   if (DataTable[index].ItemX != valueX)
       {
           Indexes.RemoveAt(index)
       }
}
```

**Comment by Miiraak** [comment]
```csharp
foreach (var item in DataProxyTable)
{
   if (item.itemX == valueX)
   {
       item.delete();
    }
}
var proxyQueue = ProxyQueuing(DataProxyTable);
return proxyQueue;
```

**Comment by Miiraak** [comment]
Patched in `[d21ebe0]` [...](https://github.com/Miiraak/ProxySharp/commit/8d03b2e53125fd8178a14c741239ea5a2048dede)
